tests/test_run_settings.py

```diff
@@ -47,7 +49,23 @@ def test_create_run_settings_returns_settings_subclasses(run_cmd, Settings):
     if _run_cmd:
         settings = create_run_settings("local", "echo", "hello", run_command=run_cmd)
         assert settings.run_command == _run_cmd
-        assert type(settings) == Settings
+        assert isinstance(settings, Settings)
+
+
+def test_create_run_settings_handles_mpiexec_settings_correctly():
+    with contextlib.ExitStack() as ctx:
+        # naively assume that if slurm is present, then the mpiexec.slurm
+        # wrapper is being used
+        if detect_launcher() == "slurm":
```

Comment:
So here's where this gets REALLY strange (and why there's that stupid help-string parsed check in the `MpiexecSettings` source code. A machine might have Slurm, but ALSO have an actual MPI library that provides a 'real' version of mpiexec. In that circumstance, this test would incorrectly fail. I think you're going to have to do a query here as well on the help strings for mpiexec to see if the `mpiexec` in the path is the Slurm one or not.

Reply:
I was afraid of that...

It's unfortunate that this test may need to dissolve into making the same check in both the application and test code, but at this point in time I don't really see a way around it.